### PR TITLE
Add release notes for 0.271

### PR DESCRIPTION
**Pull request by cocozianu**
# Missing Release Notes
## Amit Dutta
- [x] https://github.com/prestodb/presto/pull/17245 Adding checkArgument function to properly throw Presto Exception when… (Merged by: Timothy Meehan)

## Masha Basmanova
- [x] https://github.com/prestodb/presto/pull/17316 Mark tdigest_agg function as public (Merged by: Maria Basmanova)

## Rongrong Zhong
- [x] https://github.com/prestodb/presto/pull/17228 Make sure ConsistentHashingNodeProvider returns unique candidates (Merged by: Rongrong Zhong)

## singcha
- [x] https://github.com/prestodb/presto/pull/17246 add warning when map with double or real as key is created (Merged by: Rebecca Schlussel)

# Extracted Release Notes
- #17096 (Author: Arjun Gupta): Spill large blocks to separate spill files in distinct aggregates
  - Add a new configuration property ``experimental.distinct-aggregation-large-block-spill-enabled`` to enable spilling of blocks that are larger than ``experimental.distinct-aggregation-large-block-size-threshold`` bytes into a separate spill file.  This can be overridden by ``distinct_aggregation_large_block_spill_enabled`` session property.
  - Add a new configuration property ``experimental.distinct-aggregation-large-block-size-threshold`` to define the threshold size beyond which the block will be spilled into a separate spill file.  This can be overridden by ``distinct_aggregation_large_block_size_threshold`` session property.
- #17152 (Author: v-jizhang): Test and fix cast from bigint to varchar
  - Test and fix cast from bigint to varchar.
- #17190 (Author: Chunxu Tang): Support basic timestamp in the iceberg connector
  - Support basic timestamp in the iceberg connector.
- #17237 (Author: Chunxu Tang): Remove iceberg.catalog.uri from iceberg connector configs
  - Remove the iceberg.catalog.uri config. Use hive.metastore.uri instead.
- #17258 (Author: Shashwat Arghode): Add Prepared statement sql to query lifecycle events
  - Add support for viewing expanded prepared query in Web UI.
- #17262 (Author: JySongWithZhangCe): Support ORC format caching for iceberg connector
  - Support ORC format caching module for iceberg connector.
- #17268 (Author: Otakar Trunecek): Only collect stats for primitive types in ThriftHiveMetastore
  - Fix ANALYZE TABLE for partitioned Hive tables with complex columns (array, map, struct).
  - Improve performance of ANALYZE TABLE on hive tables with complex columns.
- #17297 (Author: Rongrong Zhong): Fix CBO broadcast join reordering
  - Fix reorder joins optimization where plan might not be optimal when original build side is larger than configured ``join-max-broadcast-table-size``.

# All Commits
- 0b50dccef3fa00f010f8741a662cf0504d982905 Mark tdigest_agg function as public (Masha Basmanova)
- 73516ea7bddf16bd817d0fb481dbfda3080c54e1 Reduce memory used by primitive flat map value blocks (Sergii Druzkin)
- 246dff7b880c3a2447f9506961894ea6be22394d Avoid volatile writes for empty responses in PageBufferClient (James Petty)
- f4ce31688efc1145afc1ded544d72e4cc24071f2 Fix documentation for tdigest_agg (Masha Basmanova)
- 2da621dcdf46fe15503f290773afc8ef6d83b786 Reduce memory consumed by flat map value block (Sergii Druzkin)
- 33a3f4d52a42635191dab8cc3e9f43cbd6c3417e Add Prepared statement sql to query lifecycle events (Shashwat Arghode)
- 492adedbbf4e881c3ada28738b66fffb16b9846f Disabling flaky test (Swapnil Tailor)
- 877ae0e6af8263ac86305eb4cfa46fd47ab91433 Optimize creation of OrcDecompressor.OutputBuffer in OrcInputStream (Sergii Druzkin)
- d53b41d628c0ebec52f80949fe12d9d0cd0d222d Split ci workflow into multiple workflows (Swapnil Tailor)
- 2ff205feeb11f66269664413980955d0e8a55030 Fix CBO broadcast join reordering (Rongrong Zhong)
- 7330a0400f4979f99f74b05f10fc227df4695288 Improve SimpleTtlNodeSelector performance (Neerad Somanchi)
- 2be3158b47cc97d8e6bdbafcb1b531d5cc3ac9ec Fix long dictionary raw bytes estimate (Arunachalam Thirupathi)
- 211390bd1b56a9c65ee53cea3f869a3bfc6c4a0e add warning when map with double or real as key is created (singcha)
- 60d644fbbafad63a694f05ef6dae0a465f2d5890 Option to disable string dictionary encoding in OrcWriter (Arunachalam Thirupathi)
- 61ea2fa9dec3b5923a51b5834413314d474d7196 Use Factory method in OrcWriterOptions (Arunachalam Thirupathi)
- a20713619c73bed194882fef1341f7469dceba42 Reuse compression buffer in OrcWriter (Arunachalam Thirupathi)
- ae1dec718eb24f6aaff1f779da315c0b09d5a981 Support enums in partition columnst (Pranjal Shankhdhar)
- 6ea0fd1e374d2d65952e7f0f23415c8fcf6565dd Add thrift serde support for ClusterStats (Abhishek Aggarwal)
- aae9d9f352aff81595adb65184315f0c5039152d Support ORC format caching for iceberg connector (JySongWithZhangCe)
- a8fa57f81aeb71cb9295f42f9efba8fe7ee9ae54 Fix flaky TestClusterStatsResource (abhiseksaikia)
- dfb6fd39e2745a3b5f16ab33c1f4dcd3028d787c Use minified JS for production UI (Mayank Garg)
- 1b06b014274b053a6c75e7b4ff8aa0552f29ee5b Make coordinator index page use /v1/queryState (Mayank Garg)
- 3e8e66209e1bacf1a512c14bd2ffdadf16ae2c80 Fix query ordering comparator in QueryResource (Mayank Garg)
- a7af002182098ba5a61248edfcaaa66e5d50e489 Extend QueryStateInfo to handle Coordinator UI (Mayank Garg)
- a25592e16a99280c3bb157120513ab0411c668ff Only collect stats for primitive types in ThriftHiveMetastore (Otakar Trunecek)
- 621d42d520c2d54ba11a57050a9ca8f88e79ef0d Support basic timestamp in the iceberg connector (Chunxu Tang)
- ba19b4526dd099348a88d4e424249564c0fae029 Increasing timeout for a test to avoid it's flakyness (Swapnil Tailor)
- 8987909d1ea7597c67dc39659a2879afb74ec708 Update iceberg connector doc with hadoop catalog configs (Chunxu Tang)
- 4a9681334e3c3ad410d70c3f104d9eaa342b4d11 Improve MetadataQueryOptimizer with a threshold to limit calls to Metastore (ericyuliu)
- dd0c62116f2aebae4e677bdea2ba3e6765c1df53 Make sure ConsistentHashingNodeProvider returns unique candidates (Rongrong Zhong)
- d37ac0735773af0cd17b26fff1e2c9a9575734d8 Spill large blocks to separate spill files in distinct aggregates (Arjun Gupta)
- 003ecf913b21e4d901a8b2020e264fb6ba2626fb Test and fix cast from bigint to varchar (v-jizhang)
- 66e4f63b79a00246cd729ef57781ac6af45a1422 Remove iceberg.catalog.uri from iceberg connector configs (Chunxu Tang)
- afdd49e81779f92f42e6832503bf0f3b87d26060 Add Thrift support for ResourceGroupInfo (Zitong Wei)
- c147721f4f8cd838dff4e75c100714309fbd3f72 Plumb enum type through create table/view and insert into DDL (mengdilin)
- 28abe810d70197568d6b6c2fe8de4eb252cc4d87 Adding checkArgument function to properly throw Presto Exception when argument is invalid. (Amit Dutta)

### CODE REVIEW

File: presto-docs/src/main/sphinx/release/release-0.271.rst [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+Hive Changes
+____________
+* Fix ANALYZE TABLE for partitioned Hive tables with complex columns (array, map, struct).
+* Improve performance of ANALYZE TABLE on hive tables with complex columns.
+
+Iceberg Changes
+_______________
+* Remove the iceberg.catalog.uri config. Use hive.metastore.uri instead.
+* Support ORC format caching module for iceberg connector.
```
Comment (rschlussel) [inline presto-docs/src/main/sphinx/release/release-0.271.rst:26]: Add support for

File: presto-docs/src/main/sphinx/release/release-0.271.rst [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+* Support basic timestamp in the iceberg connector.
```
Comment (rschlussel) [inline presto-docs/src/main/sphinx/release/release-0.271.rst:27]: Add support for

File: presto-docs/src/main/sphinx/release/release-0.271.rst [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+**Details**
+===========
+
+General Changes
+_______________
+* Fix reorder joins optimization where plan might not be optimal when original build side is larger than configured ``join-max-broadcast-table-size``.
+* Add a new configuration property ``experimental.distinct-aggregation-large-block-size-threshold`` to define the threshold size beyond which the block will be spilled into a separate spill file.  This can be overridden by ``distinct_aggregation_large_block_size_threshold`` session property.
+* Add a new configuration property ``experimental.distinct-aggregation-large-block-spill-enabled`` to enable spilling of blocks that are larger than ``experimental.distinct-aggregation-large-block-size-threshold`` bytes into a separate spill file.  This can be overridden by ``distinct_aggregation_large_block_spill_enabled`` session property.
+* Add support for viewing expanded prepared query in Web UI.
+* Test and fix cast from bigint to varchar.
```
[the remark this answers is not in the document]
Comment (cocozianu) [reply presto-docs/src/main/sphinx/release/release-0.271.rst]: Added an example.

File: presto-docs/src/main/sphinx/release/release-0.271.rst [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+* Add support for generating a warning when creating a map with double/real as keys.
```
[the remark this answers is not in the document]
Comment (cocozianu) [reply presto-docs/src/main/sphinx/release/release-0.271.rst]: fixed

File: presto-docs/src/main/sphinx/release/release-0.271.rst (hunk 2 above)
Comment (cocozianu) [reply presto-docs/src/main/sphinx/release/release-0.271.rst]: fixed

**Review by rschlussel — APPROVED** [review]

**Pull request merged**